app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -62,10 +62,10 @@ def reload_answer
   def publish_all
     graded_submissions = @assessment.submissions.with_graded_state
     if !graded_submissions.empty?
-      graded_submissions.update_all(
-        workflow_state: 'published', publisher_id: current_user.id, published_at: Time.zone.now,
-        updated_at: Time.zone.now, updater_id: current_user.id
-      )
+      graded_submissions.each do |submission|
```

Comment:
We need to use the `publish` and `save` callbacks because there is some logic to update stuff. Either way there is a need to do some processing on the data then save. Is there another way to do the update for this?